sqrl-openai/src/main/java/com/datasqrl/openai/RetryUtil.java

```diff
@@ -0,0 +1,38 @@
+package com.datasqrl.openai;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Optional;
+import java.util.concurrent.Callable;
+
+public class RetryUtil {
+
+    private static final Logger logger = LoggerFactory.getLogger(RetryUtil.class);
+
+    private static final String MAX_RETRIES = "FUNCTION_MAX_RETRIES";
+
+    public static <T> T executeWithRetry(Callable<T> task) {
+        int maxRetries = Optional.ofNullable(System.getenv(MAX_RETRIES))
+                .map(Integer::parseInt)
+                .orElse(3);
```

Comment:
make this a static variable `MAX_RETRIES_DEFAULT` so one doesn't have to look around the code to find the default value.